From ee4d17e4d885ee0801be7ca46f07f34cab205ec3 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Fri, 17 Dec 2010 10:54:26 +0000
Patch-mainline: 2.6.38
References: fate#310916
Git-commit: 9c3d2f7ffac34c62fea0b73e607707168a6f09b1
Subject: [PATCH 1948/2588] drm/i915: Enable EI mode for RCx decision making
 on Sandybridge

And no I have no idea what the difference is either, just that is the
recommendation.

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit 9c3d2f7ffac34c62fea0b73e607707168a6f09b1)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_display.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 3e468e6..b38978a 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -6158,6 +6158,7 @@ static void gen6_enable_rc6(struct drm_i915_private *dev_priv)
 	I915_WRITE(GEN6_RC_CONTROL,
 		   GEN6_RC_CTL_RC6p_ENABLE |
 		   GEN6_RC_CTL_RC6_ENABLE |
+		   GEN6_RC_CTL_EI_MODE(1) |
 		   GEN6_RC_CTL_HW_ENABLE);
 
 	I915_WRITE(GEN6_RC_NORMAL_FREQ,
-- 
1.7.6

